### ISSUE TITLE
VJOY PlugIn - Rework of Press/Release System OR Add a New 3rd Option

### ISSUE THREAD

**Issue by MobiFlight-Admin** [MobiFlight/MobiFlight-Connector#208]
## Problem summary ##
Sending a "momentary" Press/Release Combo instead of a fixed State

## MF Version ##
7.5.2

## Steps to reproduce ##
* Create a Input... Typ Buttom
* Install VJoy Plattform.... Define a Virtual Joystick so we can work with it in MF
* Devine the Vjoy Button to your Input Device and choose different Logics for testing

## Actual result ##
All Explaned here:   https://www.mobiflight.com/forum/topic/1788.html

## Further details ##
@ SEBASTIAN
I Still test your recomendations.... But they wont work !

If i choose On Press the ON Command ( without using the OFF Command) then the first button Press is detected in FSUIPC ( And Also in the Vjoy Oberserver tool) as a Joystick PRESS. .... But then a additional Buttenpress do nothing   ( FSUIPC and also the Vjoy Observer show the button as "still pressed" and a new command is only send if i release the button meanwhile by a direct RELEASE Command.
)
SAME Result with your recomendation to test it "reversed" with the RELEASE Command ONLY.
Here nothing happed.....  The Button MUST be Pressed already by Vjoy before i can release it.

Summary.... Whatever a couple of ON in a row or OFF in a row will NOT work.

## Request ##
If Danny is still active.... May he can reply with some ideas.
Maby my requested "Temporary Press/Release " Option as a 3rd alternate is the Key....
OR 
He can change the Code in a way (Like Sebastian recommend)  that a Command that is already Active in Vjoy ( Like a Press if Button is still pressed) occure Then in a short Release and another Press INSTEAD to the current situation where nothing happen.
( Or inverted that a Release will ocure in a short Press when i try to release it again.

Thank you guys!

[Issue created by Pizman82: 2019-03-18]
[Last updated on bitbucket: 2021-02-23]

**Comment by NickClark** [comment]
I'd love to see this happen as well. I have my panel setup for MSFS but thought I would also be able to use it as a generic HID for other sims as well via the vJoy support. Was a little startled when I was able to assign everything but my encoders. I had assumed they would work as just buttons. So I'm blocked on this last part. 

Enabling button presses for encoders would allow the panel to work for any program.

One other side thought: Instead of using vJoy, it would be cool if the board could just enumerate itself as a joystick device as well. I have VKB flight controls, and while they are joysticks by default, you can enable keyboard and mouse modes too, in which case the USB interface see the single device as a joystick, mouse, and/or keyboard.

Anyways, just some thoughts. At this point, I welcome any suggestions on how I can work around this issue. Only idea I have so far is to find some software that just programs the device as a generic game controller, then map those command back into mobiflight using its ability to use normal controllers as input devices

**Comment by DocMoebiuz** [comment]
Superseded by #1274 - we are talking about supporting Encoders... not special onrelease behavior for buttons.
